contract audit fixes

tgrecojs opened the pull request:
Refs: #141, #142

tgrecojs committed:
feat: added code for enforcing correct offerArgs shape

amessbee commented on contract/src/airdrop.contract.js:
```diff
@@ -23,6 +23,18 @@ import { objectToMap } from './helpers/objectTools.js';
 import { getMerkleRootFromMerkleProof } from './merkle-tree/index.js';
 import '@agoric/zoe/exported.js';
 
+const ProofDataShape = harden({
+  hash: M.string(),
+  direction: M.string(),
+});
+
+const OfferArgsShape = harden({
+  tier: M.number(),
+  address: M.string(),
+  key: M.string(),
+  proof: M.arrayOf(ProofDataShape),
+});
```
This looks good. Maybe we can also specify max length as (perhaps limits parameter) for `string` and `arrayOf`. WDYT?

dckc replied:
They have built-in limits.

tgrecojs replied:
I think that sounds great! Great suggestion @amessbee.

Quick note: #141 is going to eliminate the address argument as the work done for that issue computes the address from the pubkey rather than relying on the user to supply the address.

dckc replied:
oh... though max length makes sense for `address`, since the app actually knows the lengh.

maybe `proof` has an app-specified max length too; I dunno

amessbee commented on contract/src/airdrop.contract.js:
```diff
@@ -23,6 +23,18 @@ import { objectToMap } from './helpers/objectTools.js';
+});
+
+const OfferArgsShape = harden({
+  tier: M.number(),
```
`tier` seems to be defined as `bigInt` in the contract and not as a `number` - not sure if that matters.

tgrecojs replied:
ATPIT the tier argument is working  correctly within the contract. That said, the `tier` value is going to be concatenated onto the end of the pubkey hex string rather than being its own argument.

tgrecojs committed:
feat: codespace config

tgrecojs committed:
feat(depositFacet-claim): working implementation of using depositFacet to claim

tgrecojs committed:
chore: added depositFacet.test.js file

tgrecojs committed:
test(depositFacet): added test code for verifying that offers are being handled correctly when processed through updated offerHandler code

tgrecojs committed:
test: added code further usage of , and demonstrates that its well-suitedness as generic interface for creating AsyncGenerator subscribtions (#150)

This commit contains test for:
- Handling bobs offer
- Subscribing to bobs wallet
- Verifying Alice's inability to make a second claim attempt

tgrecojs committed:
Merge branch '#141-patch' into 142-offerArgs-validation

tgrecojs commented:
closing this PR as these fixes were merged in PR #149.
